src/core/retargeting_engine/python/interface/retargeter_core_types.py

```
@@ -41,39 +114,86 @@ def name(self) -> str:
         pass
 
     @abstractmethod
-    def _compute_without_context(self, inputs: RetargeterIO) -> RetargeterIO:
+    def _allocate_outputs(self) -> RetargeterIO:
+        """
+        Allocate and return an empty output tensor group dict.
+
+        Used by callers of ``_compute_without_cache`` that do not have
+        pre-existing output buffers (e.g. ``__call__``, ``_compute_with_cache``).
+        """
+        pass
```

Ruff (0.15.2)

[warning] Unnecessary `pass` statement

Remove unnecessary `pass`

(PIE790)